modules/fsi/src/kernels/AcousticInertia.C

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AcousticInertia.h"
+#include "SubProblem.h"
+#include "TimeIntegrator.h"
+#include "NonlinearSystemBase.h"
+
+registerMooseObject("FsiApp", AcousticInertia);
+
+InputParameters
+AcousticInertia::validParams()
+{
+  InputParameters params = TimeKernel::validParams();
+  params.addClassDescription("Calculates the residual for the inertial force "
+                             "which is the double time derivative of pressure.");
+  params.set<bool>("use_displaced_mesh") = true;
+  params.addParam<MaterialPropertyName>(
+      "inv_co_sq", "inv_co_sq", "Inverse of sqaure of the fluid's speed of sound.");
+  return params;
+}
+
+AcousticInertia::AcousticInertia(const InputParameters & parameters)
+  : TimeKernel(parameters),
+    _inv_co_sq(getMaterialProperty<Real>("inv_co_sq")),
+    _time_integrator(*_sys.getTimeIntegrator())
+{
+  _u_dot_old = &(_var.uDotOld());
+  _du_dot_du = &(_var.duDotDu());
+  _du_dotdot_du = &(_var.duDotDotDu());
+
+  addFEVariableCoupleableVectorTag(_time_integrator.uDotFactorTag());
+  addFEVariableCoupleableVectorTag(_time_integrator.uDotDotFactorTag());
+
+  _u_dot_factor = &_var.vectorTagValue(_time_integrator.uDotFactorTag());
+  _u_dotdot_factor = &_var.vectorTagValue(_time_integrator.uDotDotFactorTag());
+}
```

Comment:
```suggestion
AcousticInertia::AcousticInertia(const InputParameters & parameters)
  : TimeKernel(parameters),
    _inv_co_sq(getMaterialProperty<Real>("inv_co_sq")),
    _u_dot_old(_var.uDotOld()),
    _du_dot_du(_var.duDotDu()),
    _du_dotdot_du(_var.duDotDotDu()),
    _u_dot_factor(_var.vectorTagValue(_sys.getTimeIntegrator()->uDotFactorTag())),
    _u_dotdot_factor(_var.vectorTagValue(_sys.getTimeIntegrator()->uDotDotFactorTag()))
{
  addFEVariableCoupleableVectorTag(_sys.getTimeIntegrator()->uDotFactorTag());
  addFEVariableCoupleableVectorTag(_sys.getTimeIntegrator()->uDotDotFactorTag());
}
```

No need for any of these to be pointers.

Also removed the time integrator ref - it's only used in the constructor